q2_feature_table/_group.py

```diff
@@ -0,0 +1,71 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2017, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import biom
+import qiime2
+import numpy as np
+import pandas as pd
+
+
+def _collapse_factory(function):
+    def collapse_f(table, axis):
+        # axis is always the transpose of the original collapse axis
+        return np.array([function(x) for x in table.iter_data(axis=axis)])
+    return collapse_f
+
+
+_mode_lookup = {
+    'sum': _collapse_factory(np.sum),
+    'median-ceiling': _collapse_factory(lambda x: np.ceil(np.median(x))),
```

Comment:
I'm a little torn here. It feel like it would make more sense for the median to pick the "right-hand" value in the event of an even number of ids. Then all of the observation frequencies in the collapsed table are strictly values that were found in the original. (Why introduce a mean of two numbers only to take the ceiling of it? It just invents a number out of thin air...)

Apparently any value between and include the two middle values is an adequate definition of median as it minimizes the distance between all other values: https://math.stackexchange.com/a/47806. So while the above implementation isn't wrong, it seems like there's a simpler and more reasonable definition of median (pick the right hand number) that maintains closure over `FeatureTable[Frequency]`.

Reply:
Did your discussion with @gregcaporaso resolve this?

Reply:
No this came afterwards, just wanted to bring this up, what we have isn't wrong so we can run with it.